### ISSUE TITLE
Neighbor sampler behaves differently in CPU and GPU

### ISSUE THREAD

**Issue by RManLuo** [dmlc/dgl#4583]
## 🐛 Bug

I try to use `dgl.dataloading.MultiLayerFullNeighborSampler` to sample the blocks for a set of `seed_nodes` which contains the duplicated items. If I sample them in CPU, the returned mfg would contain inconsistent results as shown in #4512. However, if I sample them in GPU, the duplicated seed nodes would not been removed. 

## To Reproduce

``` python
import torch
import dgl

src = torch.LongTensor(
    [0, 0, 0, 1, 2, 2, 2, 3, 3, 4, 4, 5, 5, 6, 7, 7, 8, 9, 10,
     1, 2, 3, 3, 3, 4, 5, 5, 6, 5, 8, 6, 8, 9, 8, 11, 11, 10, 11])
dst = torch.LongTensor(
    [1, 2, 3, 3, 3, 4, 5, 5, 6, 5, 8, 6, 8, 9, 8, 11, 11, 10, 11,
     0, 0, 0, 1, 2, 2, 2, 3, 3, 4, 4, 5, 5, 6, 7, 7, 8, 9, 10])
g = dgl.graph((src, dst))

sampler = dgl.dataloading.MultiLayerFullNeighborSampler(2)

# Sample in CPU
idx = torch.LongTensor([8,8])
src_nodes, dst_nodes, mfgs = sampler.sample_blocks(g, idx)
print(dst_nodes) # tensor([8, 8])
print(mfgs[-1].num_dst_nodes()) # 1
print(mfgs[-1].dstdata) # {'_ID': tensor([8, 8])}
# Inconsistant

# Sample in GPU
device = torch.device('cuda:0')
src_nodes, dst_nodes, mfgs = sampler.sample_blocks(g.to(device), idx.to(device))
print(dst_nodes) # tensor([8, 8], device='cuda:0')
print(mfgs[-1].num_dst_nodes()) # 2
print(mfgs[-1].dstdata) # {'_ID': tensor([8, 8], device='cuda:0')}
# Consistant
```

## Expected behavior

Sample results at different devices should be the same.

## Environment

 - DGL Version (e.g., 1.0): dgl-cuda11.3 0.9.0
 - Backend Library & Version (e.g., PyTorch 0.4.1, MXNet/Gluon 1.3): pytorch 1.11.0
 - OS (e.g., Linux): Linux
 - How you installed DGL (`conda`, `pip`, source): conda
 - Build command you used (if compiling from source):
 - Python version: 3.8
 - CUDA/cuDNN version (if applicable): cuda 11.3 cudnn 8.2.0_0
 - GPU models and configuration (e.g. V100): RTX3090
 - Any other relevant information:
